beacon-chain/rpc/eth/beacon/handlers_test.go

```diff
@@ -4754,3 +4755,213 @@ func Test_validateBlobSidecars(t *testing.T) {
 	require.NoError(t, err)
 	require.ErrorContains(t, "could not verify blob proof: can't verify opening proof", s.validateBlobSidecars(b, [][]byte{blob[:]}, [][]byte{proof[:]}))
 }
+
+func TestGetPendingDeposits(t *testing.T) {
+	config := params.BeaconConfig()
+	st, _ := util.DeterministicGenesisStateElectra(t, config.MaxValidatorsPerCommittee)
+
+	validators := st.Validators()
+	deps := make([]*eth.PendingDeposit, 20)
+	for i := 0; i < len(deps); i += 1 {
+		dummySig := make([]byte, 96)
+		for j := 0; j < 96; j++ {
+			dummySig[j] = byte(i)
+		}
+		deps[i] = &eth.PendingDeposit{
+			PublicKey:             validators[i].PublicKey,
+			WithdrawalCredentials: validators[i].WithdrawalCredentials,
+			Amount:                uint64(helpers.ActivationExitChurnLimit(1_000*1e9)) / 10,
+			Slot:                  0,
+			Signature:             dummySig,
+		}
+	}
+	require.NoError(t, st.SetPendingDeposits(deps))
+
+	chainService := &chainMock.ChainService{
+		Optimistic:     false,
+		FinalizedRoots: map[[32]byte]bool{},
+	}
+	server := &Server{
+		Stater: &testutil.MockStater{
+			BeaconState: st,
+		},
+		OptimisticModeFetcher: chainService,
+		FinalizationFetcher:   chainService,
+		BeaconDB:              nil,
+		ChainInfoFetcher:      chainService,
```

Comment:
I don't think this is needed

Reply:
Good point, I only need OptimisticModeFetcher and FinalizationFetcher